recipes/7zip/19.00/conanfile.py

```diff
@@ -36,26 +39,36 @@ def _uncompress_7z(self, filename):
         self.run("lzma920\\7zr.exe x {}".format(filename))
 
     _msvc_platforms = {
-        'x86_64': 'x64',
-        'x86': 'x86',
+        "x86_64": "x64",
+        "x86": "x86",
     }
 
     def _build_msvc(self):
-        env_build = VisualStudioBuildEnvironment(self)
-        with tools.environment_append(env_build.vars):
-            vcvars = tools.vcvars_command(self.settings)
-            with tools.chdir("CPP/7zip"):
-                self.run("%s && nmake /f makefile PLATFORM=%s" % (
-                vcvars, self._msvc_platforms[str(self.settings.arch_build)]))
+        with tools.vcvars(self.settings):
+            with tools.chdir(os.path.join("CPP", "7zip")):
+                self.run("nmake /f makefile PLATFORM=%s" % (
+                self._msvc_platforms[str(self.settings.arch_build)]))
 
     def _build_autotools(self):
         # TODO: Enable non-Windows methods in configure
-        env_build = AutoToolsBuildEnvironment(self)
-        with tools.environment_append(env_build.vars):
-            with tools.chdir("CPP/7zip/Bundles/LzmaCon"):
-                self.run("make -f makefile.gcc all")
+        autotools = AutoToolsBuildEnvironment(self)
+        extra_env = {}
+        if self.settings.os_build == "Windows" and self.settings.compiler == "gcc":
+            extra_env["IS_MINGW"] = "1"
```

Comment:
I'd add here a link to the issue https://github.com/conan-io/conan-center-index/issues/394, in the future we should fix it in Conan (https://github.com/conan-io/conan/issues/1706) and come back to the recipe to remove this patch.
```suggestion
            extra_env["IS_MINGW"] = "1"  # FIXME: https://github.com/conan-io/conan/issues/1706
```